src/site/antora/modules/ROOT/pages/faq.adoc

```diff
@@ -17,404 +17,196 @@ Licensed to the Apache Software Foundation (ASF) under one or more
+* The https://www.slf4j.org/api/org/slf4j/spi/LocationAwareLogger.html#log(org.slf4j.Marker,java.lang.String,int,java.lang.String,java.lang.Object%5B%5D,java.lang.Throwable)[`org.slf4j.spi.LocationAwareLogger::log`] method is not yet implemented in a garbage-free manner in the `log4j-slf4j-impl` and `log4j-slf4j2-impl` bridges.
 It creates a new message object for each call.
 
 [#gc-free-domain-object]
 == How do I log my domain object without creating garbage?
 
-One option is to let the domain object implement java.lang.CharSequence.
+One option is to let the domain object implement `CharSequence`.
 However, for many domain objects it may not be trivial to implement this without allocating temporary objects.
 
 An alternative is to implement the `org.apache.logging.log4j.util.StringBuilderFormattable` interface.
-If an object is logged that implements this interface, its `formatTo` method is called instead of `toString()`.
-
-[source,java]
-----
-package org.apache.logging.log4j.util;
-public interface StringBuilderFormattable {
-    /**
-     * Writes a text representation of this object into the specified {@code StringBuilder},
-     * ideally without allocating temporary objects.
-     *
-     * @param buffer the StringBuilder to write into
-     */
-     void formatTo(StringBuilder buffer);
-}
-----
+If an object is logged that implements this interface, its `formatTo(StringBuilder)` method is called instead of `toString()`.
 
 [#logger-wrapper]
-== How do I create a custom logger wrapper that shows the correct class, method and line number?
+== How do I create a custom logger wrapper that shows the correct class, method, and line number?
 
 Log4j remembers the fully qualified class name (FQCN) of the logger and uses this to walk the stack trace for every log event when configured to print location.
-(Be aware that logging with location is slow and may impact the performance of your application.)
+
+[WARNING]
+====
+Be aware that logging with location is slow and may impact the performance of your application.
+====
 
 The problem with custom logger wrappers is that they have a different FQCN than the actual logger, so Log4j can't find the place where your custom logger was called.
 
 The solution is to provide the correct FQCN.
 The easiest way to do this is to let Log4j generate the logger wrapper for you.
 Log4j comes with a Logger wrapper generator tool.
-This tool was originally meant to support custom log levels and is documented https://logging.apache.org/log4j/2.x/manual/customloglevels.html#CustomLoggers[here].
-
+This tool was originally meant to support custom log levels and is documented xref:manual/customloglevels.adoc#CustomLoggers[here].
```

Comment:
The tool is currently here: apache/logging-log4j-transform#92

Reply:
:+1: `customloglevels.adoc` will be revamped in #2530.